libp2p/pubsub/pubsub_notifee.py

```diff
@@ -41,7 +49,14 @@ def __init__(self, initiator_peers_queue: "asyncio.Queue[ID]") -> None:
         await self.initiator_peers_queue.put(conn.muxed_conn.peer_id)
 
     async def disconnected(self, network: INetwork, conn: INetConn) -> None:
-        pass
+        """
+        Add peer_id to dead_peers_queue, so that pubsub and its router can
+        remove this peer_id and close the stream inbetween.
+
+        :param network: network the connection was opened on
+        :param conn: connection that was opened
+        """
+        await self.dead_peers_queue.put(conn.muxed_conn.peer_id)
```

Comment:
Can I understand `handle_dead_peer_queue` is called twice when a peer disconnects us?

Reply:
Yes, `_handle_dead_peer` might be called twice. Once when we disconnect from peer(`network` notify `pubsub`) and once when pubsub detect that the connection has closed.